File: TBD/QL/src/ql/ast/type/QLBoolean.java
package ql.ast.type;

import ql.ast.ASTNode;

/**
 * Created by Erik on 7-2-2017.
 */
public class QLBoolean implements ASTNode {
    private boolean qlBoolean;

    public QLBoolean(boolean qlBoolean) {
        this.qlBoolean = qlBoolean;
    }
}

File: WeekendWarriors/DSL/DSL/AST/Types/QLBoolean.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DSL.SemanticAnalysis.SemenaticAnalysisEvents;
using System.Diagnostics;

namespace DSL.AST
{
    public class QLBoolean : IQLExpression
    {
        public QLBoolean(bool value)
        {
            this.Value = value;
        }

        public bool Value
        {
            get;
        }

        public QLType? CheckTypes(List<QLType> parameters, QLContext context, List<ISemenaticAnalysisEvent> events)
        {
            Trace.Assert(parameters.Count == 0);
            return QLType.Bool;
        }        
    }
}
